src/components/UserInput.js

```diff
@@ -111,19 +111,22 @@ class UserInput extends Component {
           <SendIcon onClick={this._submitText.bind(this)} />
         </div>
       );
+    } else if (this.props.showFilePicker) {
+      return (
+        <div className="sc-user-input--button">
+          <FileIcon onClick={this._showFilePicker.bind(this)} />
+          <input
+            type="file"
+            name="files[]"
+            multiple
+            ref={(e) => { this._fileUploadButton = e; }}
+            onChange={this._onFilesSelected.bind(this)}
+          />
+        </div>
+      );
+    } else {
+      return null;
```

Comment:
Hm so this makes the emoji picker icon jump to the left whenever the user starts typing; I think it would be better to always render either the file picker or the send icon. The handler _submitText already just ignores empty messages so I think it would be fine.